library/bigip_user.py

```diff
@@ -182,411 +172,366 @@
+        or already processed) and contains nameReference
+        key, delete it and append the remaining dictionary element into
+        a list.
+        If the nameReference key is removed just append the dictionary
+        into the list.
 
-    def can_not_login_with_new_credentials(self):
-        try:
-            kwargs = dict(
-                server=self.params['server'],
-                user=self.params['username_credential'],
-                password=self.params['password_credential'],
-                port=self.params['server_port']
-            )
-            self.connect_to_bigip(**kwargs)
-            return False
-        except Exception:
-            return True
+        :returns list of dictionaries
 
-    def read_user_information(self):
-        user = self.load_user()
-        return self.format_user_information(user)
+        """
+        if self._values['partition_access'] is None:
+            return
+        result = []
+        part_access = self._values['partition_access']
+        for access in part_access:
+            if isinstance(access, dict):
+                if 'nameReference' in access:
+                    del access['nameReference']
+
+                    result.append(access)
+                else:
+                    result.append(access)
+            if isinstance(access, str):
+                acl = access.split(':')
+                value = dict(
+                    name=acl[0],
+                    role=acl[1]
+                )
 
-    def format_user_information(self, user):
-        result = dict()
-        result['name'] = str(user.name)
-        if hasattr(user, 'description'):
-            result['full_name'] = str(user.description)
-        if hasattr(user, 'shell'):
-            result['shell'] = str(user.shell)
-        if hasattr(user, 'partitionAccess'):
-            result['partition_access'] = self.format_current_partition_access(user)
+                result.append(value)
         return result
 
-    def format_current_partition_access(self, user):
-        result = set()
-        for acl in user.partitionAccess:
-            access = '%s:%s' % (acl['name'], acl['role'])
-            result.update(access)
-        return list(result)
+    def to_return(self):
+        result = {}
+        for returnable in self.returnables:
+            result[returnable] = getattr(self, returnable)
+        result = self._filter_params(result)
+        return result
 
-    def load_user(self):
-        if self.is_version_less_than_13():
-            return self.load_user_without_partition()
-        else:
-            return self.load_user_with_partition()
+    def api_params(self):
+        result = {}
+        for api_attribute in self.api_attributes:
+            if api_attribute in self.api_map:
+                result[api_attribute] = getattr(
+                    self, self.api_map[api_attribute])
+            elif api_attribute == 'password':
```

Comment:
you cannot do that, i have already tried that, this causes bigip password to be written here if i remove this